src/app/AttributeCache.cpp

```diff
@@ -46,6 +57,34 @@ CHIP_ERROR AttributeCache::UpdateCache(const ConcreteDataAttributePath & aPath,
         handle.RightSize();
 
         state.Set<System::PacketBufferHandle>(std::move(handle));
+
+        //
+        // Clear out the committed data version and only set it again once we have received all data for this cluster.
+        // Otherwise, we may have incomplete data that looks like it's complete since it has a valid data version.
+        //
+        mCache[aPath.mEndpointId][aPath.mClusterId].mCommittedDataVersion.ClearValue();
+
+        // This commits a pending data version if the last report path is valid and it is different from the current path.
+        if (mLastReportDataPath.IsValidConcreteClusterPath() && mLastReportDataPath != aPath)
+        {
+            CommitPendingDataVersion();
+        }
+
+        bool foundEncompassingWildcardPath = false;
+        for (const auto & path : mRequestPathSet)
+        {
+            if (path.IncludesAllAttributesInCluster(aPath))
+            {
+                foundEncompassingWildcardPath = true;
```

Comment:
```suggestion
                foundEncompassingWildcardPath = true;
                break;
```
because there's no reason to keep going at that point...  I guess @mrjerryjohns asked for the break to be removed?  But I'm not following the logic for why it's being removed.

Reply:
The break has to be present here.

Reply:
My suggestion was to avoid having a large chunk of actual logic along-side the break at the very end (which is difficult to spot), vs. just setting a flag, and then acting on that flag later.

Reply:
thanks, updated.